model-server/src/main/resources/org/modelix/model/server/store/ignite.xml

```diff
@@ -38,43 +38,19 @@
         <property name="password" value="${jdbc.pw}" />
     </bean>
 
-    <bean class="org.apache.ignite.cache.QueryEntity">
-        <property name="keyType" value="java.lang.String"/>
-        <property name="valueType" value="java.lang.String"/>
-        <property name="indexes">
-            <list>
-                <bean class="org.apache.ignite.cache.QueryIndex">
-                    <property name="name" value="pk_key"/>
-                    <property name="indexType">
-                        <util:constant static-field="org.apache.ignite.cache.QueryIndexType.SORTED"/>
-                    </property>
-                    <property name="fields">
-                        <map>
-                            <entry key="key" value="true"/>
-                        </map>
-                    </property>
-                </bean>
-            </list>
-        </property>
-    </bean>
-
     <bean id="ignite.cfg" class="org.apache.ignite.configuration.IgniteConfiguration">
-        <property name="peerClassLoadingEnabled" value="true"/>
-
+        <property name="consistentId" value="modelix-postgres"/>
```

Comment:
> Sets consistent globally unique node ID which survives node restarts.

This might break starting multiple model-server in the same cluster.
That would break rolling deployments and disable https://issues.modelix.org/issue/MODELIX-344.

I think it will break because I guess that each node in the cluster needs a "globally unique node ID".
I am going to test it.

Reply:
```
SEVERE: Failed to start manager: GridManagerAdapter [enabled=true, name=o.a.i.i.managers.discovery.GridDiscoveryManager]
class org.apache.ignite.IgniteCheckedException: Failed to start SPI: TcpDiscoverySpi [addrRslvr=null, addressFilter=null, sockTimeout=5000, ackTimeout=5000, marsh=JdkMarshaller [clsFilter=org.apache.ignite.marshaller.MarshallerUtils$1@c82d925], reconCnt=10, reconDelay=2000, maxAckTimeout=600000, soLinger=0, forceSrvMode=false, clientReconnectDisabled=false, internalLsnr=null, skipAddrsRandomization=false]
	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:280)
	at org.apache.ignite.internal.managers.discovery.GridDiscoveryManager.start(GridDiscoveryManager.java:1076)
	at org.apache.ignite.internal.IgniteKernal.startManager(IgniteKernal.java:1774)
	at org.apache.ignite.internal.IgniteKernal.start(IgniteKernal.java:1161)
	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start0(IgnitionEx.java:1725)
	at org.apache.ignite.internal.IgnitionEx$IgniteNamedInstance.start(IgnitionEx.java:1647)
	at org.apache.ignite.internal.IgnitionEx.start0(IgnitionEx.java:1089)
	at org.apache.ignite.internal.IgnitionEx.startConfigurations(IgnitionEx.java:983)
	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:889)
	at org.apache.ignite.internal.IgnitionEx.start(IgnitionEx.java:828)
	at org.apache.ignite.Ignition.start(Ignition.java:350)
	at org.modelix.model.server.store.IgniteStoreClient.<init>(IgniteStoreClient.kt:79)
	at org.modelix.model.server.store.IgniteStoreClient.<init>(IgniteStoreClient.kt:33)
	at org.modelix.model.server.Main.main(Main.kt:152)
Caused by: class org.apache.ignite.spi.IgniteSpiException: Failed to add node to topology because it has the same hash code for partitioned affinity as one of existing nodes [cacheName=ignite-sys-cache, existingNodeId=03232df7-8c18-4d79-9951-73e04c76e7b6]
	at org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi.checkFailedError(TcpDiscoverySpi.java:2110)
Caused by: class org.apache.ignite.spi.IgniteSpiException: Failed to add node to topology because it has the same hash code for partitioned affinity as one of existing nodes [cacheName=ignite-sys-cache, existingNodeId=03232df7-8c18-4d79-9951-73e04c76e7b6]

	at org.apache.ignite.spi.discovery.tcp.ServerImpl.joinTopology(ServerImpl.java:1204)
	at org.apache.ignite.spi.discovery.tcp.ServerImpl.spiStart(ServerImpl.java:472)
	at org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi.spiStart(TcpDiscoverySpi.java:2206)
	at org.apache.ignite.internal.managers.GridManagerAdapter.startSpi(GridManagerAdapter.java:277)
	... 13 more
```

It seems that setting a `consistentId` is indeed problematic.

Reply:
Removed it. Was just needed on my machine, because by default an ID is generated from the addresses of all network interface. I just have too many network interfaces, which resulted in a ID that is too long to be a valid directory name.